src/devices/bus/multibus/robotron_k7071.cpp

```diff
@@ -0,0 +1,221 @@
+// license:BSD-3-Clause
+// copyright-holders:Sergey Svishchev
+
+/*
+ * Robotron K7071 (ABS) text-only video
+ *
+ * Reference: http://www.tiffe.de/Robotron/MMS16/
+ *
+ * To do:
+ * - fix screen corruption (DMA reprogramming timing?)
+ * - loadable font - CRTC LA0 connected to NMI via gate
+ */
+
+#include "emu.h"
+#include "robotron_k7071.h"
+
+#define VERBOSE 0
+#include "logmacro.h"
+
+enum kgs_st : u8
+{
+	KGS_ST_OBF = 0x01,
+	KGS_ST_IBF = 0x02,
+	KGS_ST_INT = 0x04,
+	KGS_ST_ERR = 0x80,
+};
+
+DEFINE_DEVICE_TYPE(ROBOTRON_K7071, robotron_k7071_device, "robotron_k7071", "Robotron K7071 ABS")
+
+robotron_k7071_device::robotron_k7071_device(machine_config const &mconfig, char const *tag, device_t *owner, u32 clock)
+	: device_t(mconfig, ROBOTRON_K7071, tag, owner, clock)
+	, device_multibus_interface(mconfig, *this)
+	, m_cpu(*this, "cpu")
+	, m_dma(*this, "dma")
+	, m_crtc(*this, "crtc")
+	, m_screen(*this, "screen")
+	, m_palette(*this, "palette")
+	, m_ram(*this, "ram")
+	, m_p_chargen(*this, "chargen")
+{
+}
+
+ROM_START(robotron_k7071)
+	ROM_REGION(0x1000, "eprom", ROMREGION_ERASEFF)
+	ROM_LOAD("q209.bin", 0x0000, 0x0800, CRC(efc4aac0) SHA1(2cb4db3572e6ec7f3d3ae8e70d61b509b1fa3203))
+	ROM_LOAD("q210.bin", 0x0800, 0x0800, CRC(d23e00ed) SHA1(eaca53ed784cf263323e9abba24a364b2aadac44))
+
+	ROM_REGION(0x2000, "chargen", ROMREGION_ERASE00)
+	ROM_LOAD("q211.bin", 0x0000, 0x0800, CRC(b6241bc5) SHA1(8428e344df95451a34ad01322e245d2357712010))
+ROM_END
+
+const tiny_rom_entry *robotron_k7071_device::device_rom_region() const
+{
+	return ROM_NAME(robotron_k7071);
+}
+
+void robotron_k7071_device::device_start()
+{
+	save_item(NAME(m_kgs_datao));
+	save_item(NAME(m_kgs_datai));
+	save_item(NAME(m_kgs_ctrl));
+	save_item(NAME(m_nmi_enable));
+
+	m_bus->space(AS_IO).install_readwrite_handler(0x200, 0x203, read16s_delegate(*this, FUNC(robotron_k7071_device::io_r)), write16s_delegate(*this, FUNC(robotron_k7071_device::io_w)));
+}
+
+void robotron_k7071_device::device_reset()
+{
+	m_kgs_ctrl = KGS_ST_IBF | KGS_ST_OBF;
+	m_kgs_datao = m_kgs_datai = m_nmi_enable = 0;
+}
+
+void robotron_k7071_device::device_add_mconfig(machine_config &config)
+{
+	Z80(config, m_cpu, XTAL(16'000'000) / 6);
+	m_cpu->set_addrmap(AS_PROGRAM, &robotron_k7071_device::cpu_mem);
+	m_cpu->set_addrmap(AS_IO, &robotron_k7071_device::cpu_pio);
+
+	I8257(config, m_dma, XTAL(16'000'000) / 6);
+	m_dma->out_hrq_cb().set(FUNC(robotron_k7071_device::hrq_w));
+	m_dma->in_memr_cb().set([this] (offs_t offset) { return m_cpu->space(AS_PROGRAM).read_byte(offset); });
+	m_dma->out_iow_cb<0>().set([this] (u8 data) { m_crtc->dack_w(data); });
```

Comment:
Can you please add a `memory_access<16, 0, 0, ENDIANNESS_LITTLE>::specific` member for accessing CPU memory?  It’s a lot faster than accessing the address space like this.

Also, do you really need the trampoline for IOW -> DACK?  Can’t you just do:
```c++
	m_dma->out_iow_cb<0>().set(m_crtc, FUNC(&i8275_device::dack_w));
```